front/lib/dust_api.ts

```diff
@@ -199,6 +220,74 @@ export const DustAPI = {
 
     return _resultFromResponse(response);
   },
+
+  async getRunStatus(
+    projectId: string,
+    runId: string
+  ): Promise<DustAPIResponse<{ run: RunType }>> {
+    const response = await fetch(
+      `${DUST_API_URL}/projects/${projectId}/runs/${runId}/status`,
+      {
+        method: "GET",
+      }
+    );
+
+    return _resultFromResponse(response);
+  },
+
+  async getSpecification(
+    projectId: string,
+    specificationHash: string
+  ): Promise<
+    DustAPIResponse<{ specification: { created: number; data: string } }>
+  > {
+    const response = await fetch(
+      `${DUST_API_URL}/projects/${projectId}/specifications/${specificationHash}`,
+      {
+        method: "GET",
+      }
+    );
+
+    return _resultFromResponse(response);
+  },
+
+  async getRunBlock(
+    projectId: string,
+    runId: string,
+    runType: BlockType,
+    blockName: string
+  ): Promise<DustAPIResponse<{ run: RunType }>> {
+    const response = await fetch(
+      `${DUST_API_URL}/projects/${projectId}/runs/${runId}/blocks/${runType}/${blockName}`,
+      {
+        method: "GET",
+      }
+    );
+
+    return _resultFromResponse(response);
+  },
+
+  async createDataSource(
+    projectId: string,
+    payload: DustAPICreateDataSourcePayload
```

Comment:
Why separate with payload here vs having the different args directly?

It makes the signature a bit less understandable and you haven’t used that pattern for the other functions?

Reply:
Actually you did in the other routes.

Thinking more about it I would in-line these as this is a side effect of using an HTTP POST API that should not transpire here (you wouldn’t cut that interface if there was no network interface no?). Wdyt?

Reply:
That is fair. I did it for runs because there are 2 separate functions that use the same payload but not really useful for this one. I’ll remove!